### PR TITLE
Add general pattern serialization & matcher & RoPE

### PR DESCRIPTION
**Pull request by usstq**
### Details:
 - view LLM model in xml format inside netron is painful, `ov::pass::ModelPass` PrintModel is introduce to help, it dumps model topology into C++ pseudo-code with detailed information. this tool is not used by default, developer can register this pass into `ov::pass::Manager` at particular stage to inspect the whole ngraph topology at that stage. Both dumping & viewing in pseudo-code format is much faster,  and viewing it inside a C++ format aware editor like VSCode helps to highlight & track data flow, and almost all information contained in ngraph are exposed in this format including node's inter-connection, small constant values, shapes, attributes ...

example dumps:
```c++
    auto slice_Slice = makeOP<opset1::StridedSlice>({view_Reshape, {0, 0, 0, 0}, {0, 0, 0, 80}, {1, 1, 1, 1}}, "f32[?,?,32,80]", {{"begin_mask",  {1,1,1,0} }, {"end_mask",  {1,1,1,0} }, {"new_axis_mask",  {} }, {"shrink_axis_mask",  {} }, {"ellipsis_mask",  {} }});   //  __module.model.gpt_neox.layers.1.attention/aten::slice/Slice(__module.model.gpt_neox.layers.1.attention/aten::view/Reshape, Constant_392211, Constant_392214, Constant_392217)
    auto permute_Transpose = makeOP<opset1::Transpose>({slice_Slice, {0, 2, 1, 3}}, "f32[?,32,?,80]");   //  __module.model.gpt_neox.layers.1.attention/aten::permute/Transpose(__module.model.gpt_neox.layers.1.attention/aten::slice/Slice, 6)
    auto slice_Slice_351 = makeOP<opset1::StridedSlice>({permute_Transpose, {0, 0, 0, 0}, {0, 0, 0, 20}, {1, 1, 1, 1}}, "f32[?,32,?,20]", {{"begin_mask",  {1,1,1,0} }, {"end_mask",  {1,1,1,0} }, {"new_axis_mask",  {} }, {"shrink_axis_mask",  {} }, {"ellipsis_mask",  {} }});   //  __module.model.gpt_neox.layers.1.attention/aten::slice/Slice_351(__module.model.gpt_neox.layers.1.attention/aten::permute/Transpose, Constant_392223, Constant_392226, Constant_392229)
    auto mul_Multiply = makeOP<opset1::Multiply>({slice_Slice_351, gather_GatherElements}, "f32[?,32,?,20]", {{"auto_broadcast", "numpy"}});   //  __module.model.gpt_neox.layers.1.attention/aten::mul/Multiply(__module.model.gpt_neox.layers.1.attention/aten::slice/Slice_351, __module.model.gpt_neox.layers.1.attention/aten::gather/GatherElements)
    auto slice_Slice_420 = makeOP<opset1::StridedSlice>({slice_Slice_351, {0, 0, 0, 10}, {0, 0, 0, 2147483647}, {1, 1, 1, 1}}, "f32[?,32,?,10]", {{"begin_mask",  {1,1,1,0} }, {"end_mask",  {1,1,1,0} }, {"new_axis_mask",  {} }, {"shrink_axis_mask",  {} }, {"ellipsis_mask",  {} }});   //  __module.model.gpt_neox.layers.1.attention/aten::slice/Slice_420(__module.model.gpt_neox.layers.1.attention/aten::slice/Slice_351, ScatterUpdate_392234, Constant_392237, Constant_392240)
    auto neg_Multiply = makeOP<PowerStaticNode>({slice_Slice_420}, "f32[?,32,?,10]", {{"scale", -1.0}, {"power", 1.0}, {"shift", 0.0}, {"out-type", "f32"}});   //  __module.model.gpt_neox.layers.1.attention/aten::neg/Multiply(__module.model.gpt_neox.layers.1.attention/aten::slice/Slice_420)
    auto slice_Slice_414 = makeOP<opset1::StridedSlice>({slice_Slice_351, {0, 0, 0, 0}, {0, 0, 0, 10}, {1, 1, 1, 1}}, "f32[?,32,?,10]", {{"begin_mask",  {1,1,1,0} }, {"end_mask",  {1,1,1,0} }, {"new_axis_mask",  {} }, {"shrink_axis_mask",  {} }, {"ellipsis_mask",  {} }});   //  __module.model.gpt_neox.layers.1.attention/aten::slice/Slice_414(__module.model.gpt_neox.layers.1.attention/aten::slice/Slice_351, Constant_392290, ScatterUpdate_392292, Constant_392295)
    auto cat_Concat = makeOP<opset1::Concat>({neg_Multiply, slice_Slice_414}, "f32[?,32,?,20]", {{"axis", -1}});   //  __module.model.gpt_neox.layers.1.attention/aten::cat/Concat(__module.model.gpt_neox.layers.1.attention/aten::neg/Multiply, __module.model.gpt_neox.layers.1.attention/aten::slice/Slice_414)
    auto mul_Multiply_424 = makeOP<opset1::Multiply>({cat_Concat, gather_GatherElements_405}, "f32[?,32,?,20]", {{"auto_broadcast", "numpy"}});   //  __module.model.gpt_neox.layers.1.attention/aten::mul/Multiply_424(__module.model.gpt_neox.layers.1.attention/aten::cat/Concat, __module.model.gpt_neox.layers.1.attention/aten::gather/GatherElements_405)

```

By implementing `makeOP` support such syntax, you can just copy a snippets of the dump into your source code,  with minimum editing to generate a valid code building a ngraph sub-graph as test case.

- matching complex patterns from LLM is also painful task, the generic pattern matcher (`makePattern` ) utils is introduced to help, it supports above syntax, thus by replacing `makeOP` with `makePattern` in the dumped code, and eliminate some unwanted restrictions in attributes, it becomes a valid pattern, call `PatternValidator` inside matcher callbacks automatically validate the pattern's attributes, constant type/shape/values against matched graph.
- to make the pattern handle more general cases, we also introduced `Symbol` which represents a constant scalar with unknown value at pattern composition time, and it has overloaded C++ operator like sqrt/+/-/*/.... which allows combination into expression to appear at any place where an constant value is required in the pattern, for example:

```
    auto ndims = Symbol("ndims");
    auto view_Reshape = makePattern(ov::Rank(4));
    // view_Reshape : B,L,H,S
    auto slice_Slice_965 = GenSlice(view_Reshape, 0, ndims, 1, 3);
    auto gather_sin_cos = makePattern("f32");
    auto varsplit =
        makePattern<opset1::VariadicSplit>({gather_sin_cos, -1, {ndims/2, -1}});  //  tensor_array<f32[1,?,32] f32[1,?,32]>
```
in above code, a Symbol `ndims` is defined and used to form a Slice pattern, and symbol expression `ndims/2` is used to form VariadicSplit pattern, `PatternValidator` will collect symbol expressions appeared in the match and compute all the expressions and checks the consistency, for example, a matched Slice says the ndims is 64, then `PatternValidator` checks the matched `VariadicSplit` to see if the corresponding match of ndims/2 is really 32. and it only succeed when all symbol matches are consistent.

This greatly reduced the complexity of the pattern composition process and also makes the generated pattern more robust.

 - Rotary position embedding subgraph pattern from LLMs are matched and transformed into CPU custom RoPENode with the help of above tools. Although it's computational cost is pretty low in LLM, but it contains too many dynamic-shaped nodes and introduced considerable overheads in its `UpdateNodes` procedure, the overhead is irrelevant to the data processed but related to the number of nodes used instead, by fusing them into a single OP and also fusing preprocessing ops connecting to it (like slicing of input state tensor/gather of sin/cos cache tensor), we can effectively reduce the overheads. and the implementation is pure C++ with multi-threading parallel, so it can benefit all CPU platform.

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Comment by usstq** [comment]
@dmitry-gorokhov  @EgorDuplensky Hi this PR is another version of https://github.com/openvinotoolkit/openvino/pull/19206 with some improvements and cleanup:
  - as the former PR causes some concerns on CPU_PROFILER which I have no idea how to handle, so I remove it.
  - former PR causes concerns from Egor but he agree that it would take additional effort to implement inside CPU_DEBUG_CAP_TRANSFORMATION_SCOPE, so I still keep it as a separate MACRO inside debug_capability.hpp, and I think we can refactor these MACROS after CPU_DEBUG_CAP_TRANSFORMATION_SCOPE is refactored, so it should be ok to merge it as a temporary solution to my problem.

I really needs these tools to unblock our further development on LLM patterns , please review & comment, Thanks!

**Comment by slyalin** [comment]
I like this tool. It can be further extended to be a compilable OV graph representation in both Python and C++, not only for patterns. It may be useful for advanced developers. Does it really make sense to keep such code within openvino source tree? As I understand from @dmitry-gorokhov, it is built with an extra cmake flag, so a custom build of the whole openvino is required to use it (correct?). But it looks like we need only public OV API to build such a tool and it may not be a part of openvino itself.

**Comment by slyalin** [comment]
> I like this tool. It can be further extended to be a compilable OV graph representation in both Python and C++, not only for patterns. It may be useful for advanced developers. Does it really make sense to keep such code within openvino source tree? As I understand from @dmitry-gorokhov, it is built with an extra cmake flag, so a custom build of the whole openvino is required to use it (correct?). But it looks like we need only public OV API to build such a tool and it may not be a part of openvino itself.

I think I have found the reason to have it inside OpenVINO itself: places where we would like to generate patterns are inside the transformation pipeline, so to use this tool to dump patterns we need to modify OV and having this tool as an extra header looks simpler. If we extend it to a more wider usage, we will think about more proper packaging if it is really required.

**Comment by usstq** [comment]
@slyalin So How should I procced?

What about I change this PR into a ModelPass similar to [VisualizeTree](https://github.com/openvinotoolkit/openvino/blob/82f191b0e779e5468ac4594921027d30068335a3/src/core/include/openvino/pass/visualize_tree.hpp#L29) (also put it in the same folder), so when developers needs this, they can just call `REGISTER_PASS` to enable it to get a psuedo-code dump.

and the [gen_pattern.hpp](https://github.com/openvinotoolkit/openvino/pull/20729/files#diff-d97faa655d4d8c338df17b53bab7bca663459163c69085fecb9eed3a2f1367a1) can be placed in src/common/transformations/include/transformations/utils folder maybe?

**Comment by usstq** [comment]
Hi @slyalin @EgorDuplensky , @dmitry-gorokhov and I decide to put the gen_pattern logic inside CPU plugin first for enabling CPU optimization of RoPE, and maybe move it to ngraph later in following PR.

**Comment by yuxu42** [comment]
@EgorDuplensky Could you please continue to review? Thanks!

File: src/plugins/intel_cpu/src/utils/print_model.hpp
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/utils/print_model.hpp:1]: Is it used somewhere now? 
If not would prefer to extract in separate PR.

Comment (usstq) [reply src/plugins/intel_cpu/src/utils/print_model.hpp:1]: It's vital to enable complex pattern matching such as RoPE, it is used to dump the LLM topology into C++ pseudo-code format. Viewing LLM's topology is almost impossible w/o the help of this pass (netron.app is hopelessly slow to open IR of such big LLM), not mentioning to write any pattern based on it.

it's not used in the final code, because it's a helper pass to assist developer as [VisualizeTree](https://github.com/openvinotoolkit/openvino/blob/82f191b0e779e5468ac4594921027d30068335a3/src/core/include/openvino/pass/visualize_tree.hpp#L29) does, developer needs to manually insert it at particular stage within passManager to dump the model into C++ pseudo-code format, by insert code like following:
```
CPU_REGISTER_PASS_X64(postLPTPassManager, ov::pass::PrintModel, "rope0.cpp");
```
Then at runtime, following print will shown and file `modelprint_1_rope0.cpp` will be generated for viewing topology and extract pattern from:
```
PrintModel to modelprint_1_rope0.cpp ... done.
```

File: src/plugins/intel_cpu/src/utils/print_model.hpp [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+                os << sep << tag << iop->get_friendly_name();
+                if (iop->get_output_size() > 1) {
+                    auto out_port = vout.get_index();
+                    os << "[" << out_port << "]";
+                }
+                sep = ", ";
+            }
+            os << ")" << std::endl;
+
+            //==========================================================
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/utils/print_model.hpp:372]: Please cleam up commented code in this file

File: src/plugins/intel_cpu/src/utils/print_model.hpp [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+        m_file_name = std::string("modelprint_") + std::to_string(dump_index) + "_" + file_name;
+        dump_index++;
+    }
+    ~PrintModel() {}
+
+    bool run_on_model(const std::shared_ptr<ov::Model>& model) override {
+        if (m_file_name.empty())
+            return false;
+
+        std::cout << "PrintModel to " << m_file_name << " ... " << std::flush;
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/utils/print_model.hpp:408]: Do we really need this print? Seems like pure debug stuff

File: src/plugins/intel_cpu/src/utils/print_model.hpp [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+        std::ofstream ofs(m_file_name);
+        if (!ofs) {
+            // OPENVINO_WARN << "Error opening file " << m_file_name << " for output" << std::endl;
+            return false;
+        }
+        detail::dump_cpp_style(ofs, model);
+        ofs.close();
+        std::cout << "done." << std::endl;
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/utils/print_model.hpp:416]: Same

File: src/plugins/intel_cpu/tests/unit/ngraph_transformations/convert_to_rope.cpp
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/tests/unit/ngraph_transformations/convert_to_rope.cpp:1]: We need to find out how RoPE implementation can be tested from accuracy perspective. Right now only the transformation is covered. 
My proposal is link gen_pattern inside CPU func tests and right subgraph test ([example](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/tests/functional/subgraph_tests/src/ngram.cpp)) which is supposed to be converted into Rope inside the plugin.

File: src/plugins/intel_cpu/src/nodes/rope.cpp [hunk 4]
```diff
@@ -0,0 +1,230 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "rope.h"
+
+#include <dnnl_extension_utils.h>
```
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/nodes/rope.cpp:7]: OK

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/op/rope.hpp [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        size_t slice_start = 0;  // slice inner-most dimensions of input
+        size_t slice_stop = 0;
+        bool input_trans0213 = false;  // transpose input dim 1&2
+                                       //   from [batch, seq_length, head_cnt, head_dims]
+                                       //   into [batch, head_cnt, seq_length, head_dims]
+                                       //   before RoPE
+        bool is_interleaved = false;   // interleaved mode (for GPTJ)
+                                       //   this also implies trans0213 happens after RoPE
+        size_t ndims = 0;
+        int gather_position_arg_id = 0;
```
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/transformations/cpu_opset/common/op/rope.hpp:29]: Sure, Added

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp [hunk 6]
```diff
@@ -0,0 +1,449 @@
...
+#include <openvino/pass/pattern/op/or.hpp>
+#include <openvino/pass/pattern/op/wrap_type.hpp>
+#include <transformations/utils/utils.hpp>
+
+#include "itt.hpp"
+#include "ov_ops/type_relaxed.hpp"
+#include "transformations/cpu_opset/common/op/rope.hpp"
+#include "utils/gen_pattern.hpp"
+
+#ifdef CPU_DEBUG_CAPS
```
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp:22]: removed

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.hpp [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+public:
+    OPENVINO_RTTI("EliminateStridedSlice", "0");
+    EliminateStridedSlice();
+};
+
+class RoPEFusion : public ngraph::pass::GraphRewrite {
+public:
+    OPENVINO_RTTI("RoPEFusion", "0");
+    RoPEFusion() {
+#ifdef CPU_DEBUG_CAPS
```
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.hpp:52]: removed

File: src/plugins/intel_cpu/src/utils/print_model.hpp (hunk 1 above)
Comment (usstq) [reply src/plugins/intel_cpu/src/utils/print_model.hpp:372]: OK

File: src/plugins/intel_cpu/src/nodes/rope.cpp [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+namespace node {
+
+#define THROW_ERROR IE_THROW() << getTypeStr() << " node with name '" << getName() << "' "
+
+#if defined(OPENVINO_ARCH_X86_64)
+
+#endif  // OPENVINO_ARCH_X86_64
+
+RoPE::RoPE(const std::shared_ptr<ngraph::Node>& op, const GraphContext::CPtr context)
+    : Node(op, context, InternalDynShapeInferFactory()) {
```
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/nodes/rope.cpp:42]: OK, will change

File: src/plugins/intel_cpu/src/nodes/rope.cpp [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+            for (; i < feature_size; i++) {
+                dst[i] = src[i];
+            }
+        });
+    }
+};
+
+template <typename T>
+struct RoPEExecutorInterleaved : public RoPE::Executor {
+    void execute(RoPE* pnode) override {
```
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/nodes/rope.cpp:120]: OK, changed.

File: src/plugins/intel_cpu/src/nodes/rope.cpp [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+        if (rtPrecision == InferenceEngine::Precision::BF16) {
+            m_executor = std::make_shared<RoPEExecutorRotateHalf<ov::bfloat16>>();
+        } else {
+            m_executor = std::make_shared<RoPEExecutorRotateHalf<float>>();
+        }
+    }
+
+    // initialize input ports
+    std::vector<PortConfigurator> inPortConfigs;
+    inPortConfigs.emplace_back(LayoutType::ncsp, rtPrecision, getInputShapeAtPort(0), false, -1);
```
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/nodes/rope.cpp:193]: Sure, checked & fallbacked inside `if (m_config.is_interleaved)` block and its `else` block

File: src/plugins/intel_cpu/src/nodes/rope.h [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+    bool created() const override {
+        return getType() == Type::RoPE;
+    }
+    bool needShapeInfer() const override {
+        return false;
+    };
+    bool needPrepareParams() const override {
+        return false;
+    };
+    bool isExecutable() const override {
```
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/nodes/rope.h:33]: OK, deleted.

File: src/plugins/intel_cpu/src/nodes/rope.h [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+        return true;
+    }
+    void executeDynamicImpl(dnnl::stream strm) override {
+        execute(strm);
+    }
+    void initSupportedPrimitiveDescriptors() override;
+    void execute(dnnl::stream strm) override;
+    static bool isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept;
+
+    struct Executor {
```
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/nodes/rope.h:43]: Moved to private.

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/op/rope.hpp [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+    RoPENode() = default;
+
+    struct Config {
+        size_t slice_start = 0;  // slice inner-most dimensions of input
+        size_t slice_stop = 0;
+        bool input_trans0213 = false;  // transpose input dim 1&2
+                                       //   from [batch, seq_length, head_cnt, head_dims]
+                                       //   into [batch, head_cnt, seq_length, head_dims]
+                                       //   before RoPE
+        bool is_interleaved = false;   // interleaved mode (for GPTJ)
```
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/transformations/cpu_opset/common/op/rope.hpp:26]: OK, a detailed description is added and GPTJ is removed

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/op/rope.hpp [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+                                       //   this also implies trans0213 happens after RoPE
+        size_t ndims = 0;
```
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/transformations/cpu_opset/common/op/rope.hpp:28]: not all dimension are used for rotary embedding in some LLM, sometime only partial are used and the rest dimensions are just copied, to avoid confusion I changed the name into `rotary_ndims` and added a comment.

File: src/plugins/intel_cpu/src/utils/print_model.hpp (hunk 3 above)
Comment (usstq) [reply src/plugins/intel_cpu/src/utils/print_model.hpp:416]: removed

File: src/plugins/intel_cpu/src/utils/print_model.hpp (hunk 2 above)
Comment (usstq) [reply src/plugins/intel_cpu/src/utils/print_model.hpp:408]: removed

File: src/plugins/intel_cpu/tests/unit/ngraph_transformations/convert_to_rope.cpp
Comment (usstq) [reply src/plugins/intel_cpu/tests/unit/ngraph_transformations/convert_to_rope.cpp:1]: OK, added

File: src/plugins/intel_cpu/src/nodes/rope.h [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+        return false;
+    };
+    void executeDynamicImpl(dnnl::stream strm) override {
+        execute(strm);
+    }
+    void initSupportedPrimitiveDescriptors() override;
+    void execute(dnnl::stream strm) override;
+    static bool isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept;
+
+    const RoPENode::Config& getConfig() const {
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/rope.h:37]: minor: getConfig still seems should be in private section.

File: src/plugins/intel_cpu/src/utils/print_model.hpp (hunk 2 above)
Comment (dmitry-gorokhov) [reply src/plugins/intel_cpu/src/utils/print_model.hpp:401]: Still there

File: src/plugins/intel_cpu/src/utils/print_model.hpp (hunk 3 above)
Comment (dmitry-gorokhov) [reply src/plugins/intel_cpu/src/utils/print_model.hpp:409]: Still there

File: src/plugins/intel_cpu/src/nodes/rope.cpp (hunk 10 above)
[the remark this answers is not in the document]
Comment (dmitry-gorokhov) [reply src/plugins/intel_cpu/src/nodes/rope.cpp:155]: I still don't see any logic that changes value of rtPrecision to FP32 for unsupported use-cases. You need to initialize Port configuration with correct precision.

**Comment by EgorDuplensky** [comment]
@usstq Is it possible to split the PR into two? First one to include only pattern matching enchantments and the second one - the rest.

File: src/plugins/intel_cpu/src/nodes/rope.cpp (hunk 10 above)
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/nodes/rope.cpp:155]: Sorry, I misunderstood your point, now I see the point, changed.

File: src/plugins/intel_cpu/src/utils/print_model.hpp (hunk 2 above)
Comment (usstq) [reply src/plugins/intel_cpu/src/utils/print_model.hpp:401]: Sorry, I forgot commit the file, will remove

File: src/plugins/intel_cpu/src/utils/print_model.hpp (hunk 3 above)
Comment (usstq) [reply src/plugins/intel_cpu/src/utils/print_model.hpp:409]: Sorry, I forgot commit the file, will remove

File: src/plugins/intel_cpu/src/nodes/rope.h (hunk 15 above)
Comment (usstq) [reply src/plugins/intel_cpu/src/nodes/rope.h:37]: It's not used anymore, removed

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp [hunk 16]
```diff
@@ -0,0 +1,449 @@
...
+        rope_node->validate_and_infer_types();
+        register_new_node(rope_node);
+        return true;
+    };
+    auto m = std::make_shared<ngraph::pattern::Matcher>(result, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+// remove stridedslice from 0 to int32_max with stride 1
+ov::intel_cpu::EliminateStridedSlice::EliminateStridedSlice() {
```
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp:275]: In python, slicing a list using `a[:]` generates a new copy of the list, but in torch, if a is a tensor, then slicing generates a view referencing the old tensor w/o copy, so the StridedSlice is truly redundant and should be removed.
For now, maybe we can keep it as temporary workaround which can be removed after FE fixed this issue.

**Review by dmitry-gorokhov — APPROVED** [review]

**Comment by dmitry-gorokhov** [comment]
Currect state looks good to me. Merging.
Some follow-ups:
1. We will need to consider reuse of common RoPE op impl + transformation once ready. Would be nice to see if fusion transformatino complexity can be reduced.  https://github.com/openvinotoolkit/openvino/pull/20734
2. I bet we need fix on PT FE side to get rid of EliminateStridedSlice transformation. @usstq could you please create corresponding ticket?
3. Expose gen_pattern functionalty as common utility in trasformation library. @slyalin @jane-intel  Please provide your input to determine next steps.
